torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -1058,6 +1057,168 @@ void PropagateShapesOnBlock(Block* b, const AliasDb& db) {
   }
 }
 
+// SHAPE CACHINHG CODE
+using CanonicalArg = c10::variant<CanonicalizedSymbolicShape, IValue>;
+using CanonicalArgVec = std::vector<CanonicalArg>;
+using CanonicalRet = std::vector<CanonicalizedSymbolicShape>;
+using ShapeCacheKey = std::tuple<c10::OperatorName, CanonicalArgVec>;
+
+CanonicalArgVec cannonicalizeVec(
+    const std::vector<SSAInput>& arg_vec,
+    std::unordered_map<int64_t, int64_t>& ss_map) {
+  CanonicalArgVec canonical_args;
+  canonical_args.reserve(arg_vec.size());
+  for (auto& arg : arg_vec) {
+    if (const IValue* iv = c10::get_if<IValue>(&arg)) {
+      canonical_args.push_back(iv->deepcopy());
+    } else {
+      auto& ss = c10::get<at::SymbolicShape>(arg);
+      canonical_args.emplace_back(CanonicalizedSymbolicShape(ss, ss_map));
+    }
+  }
+  return canonical_args;
+}
+
+std::vector<CanonicalizedSymbolicShape> cannonicalizeVec(
+    const std::vector<at::SymbolicShape>& ret_vec,
+    std::unordered_map<int64_t, int64_t>& ss_map) {
+  std::vector<CanonicalizedSymbolicShape> canonical_rets;
+  canonical_rets.reserve(ret_vec.size());
+  for (auto& ss : ret_vec) {
+    canonical_rets.emplace_back(CanonicalizedSymbolicShape(ss, ss_map));
+  }
+  return canonical_rets;
+}
+
+struct ArgumentsHasher {
+  size_t operator()(const ShapeCacheKey& cacheKey) const {
+    auto& op_name = std::get<0>(cacheKey);
+    auto& arg_vec = std::get<1>(cacheKey);
+
+    // Only need operator name, as the the arguments can disambiguate the
```

Comment:
This seems fragile. Could we use the operator name + the overload name ?

Reply:
Oops, the comment I wrote is wrong. OperatorName includes the overload name in it.